sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClientIntegrationTest.java

```diff
@@ -54,4 +54,53 @@ void receiveMessageAutoComplete() {
             })
             .verifyComplete();
     }
+
+    /**
+     * Verifies that we can send and peek a message.
+     */
+    @Test
+    void peekMessage() {
+        // Arrange
+        final String messageId = UUID.randomUUID().toString();
+        final String contents = "Some-contents";
+        final ServiceBusMessage message = TestUtils.getServiceBusMessage(contents, messageId, 0);
+        final ReceiveMessageOptions options = new ReceiveMessageOptions().setAutoComplete(true);
+        receiver = createBuilder()
```

Comment:
You should dispose of the existing one (created in BeforeEach) if you are going to recreate it. It'll still consume resources.